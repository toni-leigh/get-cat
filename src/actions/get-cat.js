const GET_CAT = 'GET_CAT';

const getCat = () => ({ type: GET_CAT });

export { getCat, GET_CAT };