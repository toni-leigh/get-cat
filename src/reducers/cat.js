import { GET_CAT } from '../actions/get-cat';

const cat = (state = { imageSrc: '' }, action) => {
  switch (action.type) {
    case GET_CAT:
      return state;
    default:
      return state;
  }
}

export default cat;