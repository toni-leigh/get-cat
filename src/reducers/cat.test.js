import cat from './cat';
import { GET_CAT } from '../actions/get-cat';

test('cat reducer without state uses default', () => {
  expect(cat(undefined, { type: 'STUBBED' })).toMatchSnapshot();
});

test('GET_CAT', () => {
  const state = { imageSrc: 'foo' };
  expect(cat(state, { type: GET_CAT })).toMatchSnapshot();
});