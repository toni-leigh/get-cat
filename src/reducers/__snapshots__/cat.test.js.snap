// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET_CAT 1`] = `
Object {
  "imageSrc": "foo",
}
`;

exports[`cat reducer without state uses default 1`] = `
Object {
  "imageSrc": "",
}
`;
